### ISSUE TITLE
[Bug]: Sending author appeals for unreviewed versions rejected or disabled will not flag versions for HR

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons#15114]
### What happened?

Followup https://github.com/mozilla/addons/issues/15068 

1. I submitted a first version for an extension/theme
2. The version is rejected/disabled
3. Author is sending an appeal

### What did you expect to happen?

The version is not flagged for HR

Checking the content of the reviewer page, the appeals are present 

Example:
https://reviewers.addons.allizom.org/en-US/reviewers/review/1031754

### Is there an existing issue for this?

- [X] I have searched the existing issues

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/AMOENG-1168)

**Comment by ioanarusiczki** [comment]
I tried on -dev with listed extensions and themes and I've noticed that now they're flagged for HR after the author appeal is sent. 

What does not get flagged are the unlisted extension, some examples

Unlisted rejected -> does not get HR
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review-unlisted/634282#review-actions

Unlisted disabled -> does not get HR
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review-unlisted/634283#review-actions

Unlisted restricted -> does not get HR
 https://reviewers.addons-dev.allizom.org/en-US/reviewers/review-unlisted/634284

Another discussion would be what's happening when these appeals need to be resolved because there's reinstating content or deny the appeal. The latter should work fine but I am not sure about reinstate. I'll try to make a document with an example for each the incoming days.

Question is about unlisted, should I file a followup?
